### PR TITLE
Require `$namespace` when using `rest_url()` or `get_rest_url()`

### PR DESCRIPTION
**Pull request by danielbachhuber**
See https://github.com/WP-API/WP-API/issues/1504

### CODE REVIEW

**Comment by danielbachhuber** [comment]
@WP-API/amigos before I go about a global find and replace, are we in agreement on these changes?

**Comment by danielbachhuber** [comment]
@pento @staylor I'd be interested in your thoughts too.

File: wp-includes/rest-api/rest-functions.php [hunk 1]
```diff
@@ -226,17 +226,15 @@ function rest_get_url_prefix() {
...
+ * @param string $route    Optional. REST route. Default '/'.
  * @param string $scheme  Optional. Sanitization scheme. Default 'json'.
+ * @param int    $blog_id Optional. Blog ID. Default of null returns URL for current blog.
  * @return string Full URL to the endpoint.
  */
-function get_rest_url( $blog_id = null, $path = '/', $scheme = 'json' ) {
-	if ( empty( $path ) ) {
-		$path = '/';
-	}
+function get_rest_url( $namespace, $route = '/', $scheme = 'json', $blog_id = null ) {
```
Comment (rachelbaker) [inline wp-includes/rest-api/rest-functions.php:235]: @danielbachhuber FYI the `$scheme` default value is `rest` now.  https://core.trac.wordpress.org/ticket/34300

Comment (danielbachhuber) [inline wp-includes/rest-api/rest-functions.php:235]: Ok.

Btw, if it's not obvious, I'm changing the order of the arguments. My rationale: if we're making a breaking change, we might as well put these in the proper order.

Comment (rachelbaker) [inline wp-includes/rest-api/rest-functions.php:235]: I believe the order was originally created to match `get_home_url()`, which is why `$blog_id` is first.

Comment (danielbachhuber) [inline wp-includes/rest-api/rest-functions.php:235]: Right — but `$namespace` is a required param.

Comment (danielbachhuber) [inline wp-includes/rest-api/rest-functions.php:235]: Oh, something else I thought about. With this pull request, there's no way to `get_rest_url()` for the API index.

Given the breaking change and the fact that you can't `get_rest_url()` for the API index, I think I'm actually :-1: for this pull request.

Comment (rmccue) [inline wp-includes/rest-api/rest-functions.php:235]: `get_rest_url( '/' )` should be fine, right? That gets you the index (`route = '/'`) of the meta-namespace. Just needs a conditional here for `if ( $route !== '/' ) $path .= '/' . ...`

**Pull request closed without merging**